code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ExternalHostSchedulerUpdateEventCreate.java

```diff
@@ -0,0 +1,54 @@
+package io.cattle.platform.servicediscovery.process;
+
+import io.cattle.platform.core.model.ExternalEvent;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessHandler;
+import io.cattle.platform.process.externalevent.ExternalEventConstants;
+import io.cattle.platform.servicediscovery.deployment.DeploymentManager;
+import io.cattle.platform.servicediscovery.service.ServiceLookup;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class ExternalHostSchedulerUpdateEventCreate extends AbstractObjectProcessHandler {
+
+    @Inject
+    DeploymentManager deploymentManager;
+
+    ServiceLookup lookup;
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        ExternalEvent event = (ExternalEvent) state.getResource();
+
+        if (!ExternalEventConstants.KIND_EXTERNAL_HOST_EVENT.equals(event.getKind())
+                || !ExternalEventConstants.TYPE_SCHEDULER_UPDATE.equals(event.getEventType())) {
+            return null;
+        }
+
+        List<Service> services = new ArrayList<>();
+        Collection<? extends Service> lookupSvs = lookup.getServices(state.getResource());
+        if (lookupSvs != null) {
+            services.addAll(lookupSvs);
+        }
+        deploymentManager.reconcileServices(services);
+
+        return null;
+    }
+
+    public void setLookup(ServiceLookup lookup) {
+        this.lookup = lookup;
+    }
+
+    @Override
+    public String[] getProcessNames() {
```

Comment:
do you think it would make sense to add it to the list of service reconcile triggers instead of creating a separate file? https://github.com/rancher/cattle/blob/master/code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServicesReconcilePostTrigger.java